vendor/github.com/openshift/library-go/test/library/etcdverticalscaling/helpers.go

```diff
@@ -0,0 +1,282 @@
+package etcdverticalscaling
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/pointer"
+
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	machinev1beta1client "github.com/openshift/client-go/machine/clientset/versioned/typed/machine/v1beta1"
+)
+
+const masterMachineLabelSelector = "machine.openshift.io/cluster-api-machine-role" + "=" + "master"
+const machineDeletionHookName = "EtcdQuorumOperator"
+const machineDeletionHookOwner = "clusteroperator/etcd"
+
+type TestingT interface {
+	Logf(format string, args ...interface{})
+}
+
+// CreateNewMasterMachine creates a new master node by cloning an existing Machine resource
+func CreateNewMasterMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface) (string, error) {
+	machineList, err := machineClient.List(ctx, metav1.ListOptions{LabelSelector: masterMachineLabelSelector})
+	if err != nil {
+		return "", err
+	}
+	var machineToClone *machinev1beta1.Machine
+	for _, machine := range machineList.Items {
+		machinePhase := pointer.StringDeref(machine.Status.Phase, "Unknown")
+		if machinePhase == "Running" {
+			machineToClone = &machine
+			break
+		}
+		t.Logf("%q machine is in unexpected %q state", machine.Name, machinePhase)
+	}
+
+	if machineToClone == nil {
+		return "", fmt.Errorf("unable to find a running master machine to clone")
+	}
+	// assigning a new Name and clearing ProviderID is enough
+	// for MAO to pick it up and provision a new master machine/node
+	machineToClone.Name = fmt.Sprintf("%s-clone", machineToClone.Name)
+	machineToClone.Spec.ProviderID = nil
+	machineToClone.ResourceVersion = ""
```

Comment:
I thought we also had to remove the machine annotation and generation but maybe that doesn't matter and gets overwritten for a new machine:
```yaml
annotations:
    machine.openshift.io/instance-state: running
  ...
  generation: 2
```